### PR TITLE
[DependencyInjection] Add documentation for service locator changes

### PR DESCRIPTION
**Pull request by codedmonkey**
Adds documentation for PR https://github.com/symfony/symfony/pull/28571

### CODE REVIEW

**Comment by codedmonkey** [comment]
Code has been merged :)

**Comment by javiereguiluz** [comment]
@codedmonkey for this kind of features, we always add some comments and examples in the code samples, to make it easier to understand. Could you please my reword to do that? Thanks!

**Comment by javiereguiluz** [comment]
@codedmonkey the current YAML example that I added is wrong:

```yaml
# config/services.yaml
services:
    app.command_handler_locator:
        class: Symfony\Component\DependencyInjection\ServiceLocator
        arguments:
           -
               App\FooCommand: '@app.command_handler.foo'
               App\BarCommand: '@app.command_handler.bar'
               # if the element has no key, the ID of the original service is used
               '@app.command_handler.baz'
```

It says:

```
(<unknown>): could not find expected ':' while scanning a simple key
at line 10 column 16
```

What's the correct syntax for YAML? Or maybe this feature is not supported in YAML? Thanks!

**Comment by javiereguiluz** [comment]
Anyone knows if this feature works with YAML config? @xabbuh? @nicolas-grekas? If it works, which is the syntax? Thanks!

**Comment by codedmonkey** [comment]
My apologies for the late response.

It looks like combining collections aand arrays in YAML is indeed a no-go, https://github.com/symfony/symfony/issues/23664. It's possible to do either

```
        arguments:
           -
               - '@app.command_handler.foo'
               - '@app.command_handler.bar'
               - '@app.command_handler.baz'
```
or
```
        arguments:
           -
               App\FooCommand: '@app.command_handler.foo'
               App\BarCommand: '@app.command_handler.bar'
               App\BazCommand: '@app.command_handler.baz'
```

but not a combination of the two. Trying to add the feature would just increase complexity. Any ideas how best to address this in the docs?

**Comment by xabbuh** [comment]
That's correct, a collection in YAML is either a sequence or a mapping. I would just add a second example like this:

```yaml
# config/services.yaml
services:
    app.command_handler_locator:
        class: Symfony\Component\DependencyInjection\ServiceLocator
        arguments:
           -
               App\FooCommand: '@app.command_handler.foo'
               App\BarCommand: '@app.command_handler.bar'

    # or if the element has no key, the ID of the original service is used
    app.another_command_handler_locator:
        class: Symfony\Component\DependencyInjection\ServiceLocator
        arguments:
           -
               - '@app.command_handler.baz'
```

File: service_container/service_subscribers_locators.rst [hunk 1]
```diff
@@ -314,6 +321,10 @@ include as many services as needed in it.
     previous Symfony versions you always needed to add the
     ``container.service_locator`` tag explicitly.
 
+.. versionadded:: 4.2
```
Comment (OskarStark) [inline service_container/service_subscribers_locators.rst]: Please add a blank line after the directive

**Comment by javiereguiluz** [comment]
It took us forever ... but this is finally merged. Thank you all for your help and contributions.

**Pull request merged**